src/bam/record.jl

```diff
@@ -41,7 +41,7 @@ function Base.convert(::Type{Record}, data::Vector{UInt8})
     record = Record()
     dst_pointer = Ptr{UInt8}(pointer_from_objref(record))
     unsafe_copyto!(dst_pointer, pointer(data), FIXED_FIELDS_BYTES)
-    dsize = data_size(record)
+    dsize = record.block_size - FIXED_FIELDS_BYTES + sizeof(record.block_size)
```

Comment:
This calculation occurs more than once. It deserves its own function, which may be separate from `data_size`.